datasetinsights/estimators/faster_rcnn.py

```diff
@@ -355,619 +884,123 @@ def _gt_preds2tensor(gt_preds, max_boxes=MAX_BOXES_PER_IMAGE):
+        figsize (tuple): figure size of the plot. Default is (20, 10)
 
-        Returns:
-            saved full path of the serialized estimator
-        """
-        save_dict = {
-            "model": self.model_without_ddp.state_dict(),
-            "config": self.config,
-        }
-        torch.save(save_dict, path)
+    Returns (matplotlib.pyplot.figure): a bar plot for metric per class.
+    """
+    label_id = [k for k in sorted(data.keys()) if k in label_mappings]
+    metric_values = [data[i] for i in label_id]
+    label_name = [label_mappings[i] for i in label_id]
+    fig = plt.figure(figsize=figsize)
+    plt.bar(label_id, metric_values)
+    plt.title(f"{metric_name} per class")
+    plt.xlabel("label name")
+    plt.ylabel(f"{metric_name}")
+    plt.xticks(
+        label_id, label_name, rotation="vertical",
+    )
+    plt.margins(0.2)
+    plt.subplots_adjust(bottom=0.3)
+    return fig
 
-        return path
 
-    def load(self, path):
-        """ Load Estimator from path
+class PrepBoxes:
+    def __call__(self, image, target):
+        target = prepare_bboxes(target)
+        return image, target
 
-        Args:
-            path (str): full path to the serialized estimator
-        """
-        logger.info(f"loading checkpoint from file")
-        checkpoint = torch.load(path, map_location=self.device)
-        self.model_without_ddp.load_state_dict(checkpoint["model"])
-        loaded_config = copy.deepcopy(checkpoint["config"])
-        stored_config = copy.deepcopy(self.config)
-        del stored_config["checkpoint_file"]
-        if stored_config != loaded_config:
-            logger.debug(
-                f"Found difference in estimator config."
-                f"Estimator loaded from {path} was trained using config: \n"
-                f"{loaded_config}. \nHowever, the current config is: \n"
-                f"{self.config}."
-            )
-        self.model_without_ddp.eval()
 
-    def predict(self, pil_img, box_score_thresh=0.5):
-        """ Get prediction from one image using loaded model
+class ToTensor:
+    def __call__(self, image, target):
+        image = torchvision.transforms.functional.to_tensor(image)
+        return image, target
 
-        Args:
-            pil_img (PIL Image): PIL image from dataset.
-            box_score_thresh (float): box score threshold for filter out lower
-            score bounding boxes. Defaults to 0.5.
 
-        Returns:
-            filtered_pred_annotations (List[BBox2D]): high predicted score
-            bboxes from the model.
-        """
-        pil_img = pil_img.unsqueeze(0)
-        img_tensor = pil_img.to(self.device)
-        predicts = self.model_without_ddp(img_tensor)
-        predict_annotations = convert_bboxes2canonical(predicts)[0]
-        filtered_pred_annotations = [
-            box for box in predict_annotations if box.score >= box_score_thresh
-        ]
-        return filtered_pred_annotations
+def get_transform():
```

Comment:
Why is this a module level method? Maybe it can be a staticmethod of FasterRCNN class?

Reply:
This we can keep outside, this will be called from many places where data transformations requires like dummy data loader and others.

Reply:
The question to answer is, is this method specific for FasterRCNN? If the answer is yes, why is it in the module? Is there other types of model that will use these transformations?